perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/MembersManagerBlImpl.java

```diff
@@ -2493,53 +2493,78 @@ private boolean isMemberInGracePeriod(Map<String, String> membershipExpirationRu
 	}
 
 	@Override
-	public void sendPasswordResetLinkEmail(PerunSession sess, Member member, String namespace, String url,
+	public void sendUsernameReminderEmail(PerunSession sess, Member member, final String namespace, String mailAddress,
+										  String language) {
+
+		User user = perunBl.getUsersManagerBl().getUserByMember(sess, member);
+
+		List<Attribute> logins = perunBl.getAttributesManagerBl().getLogins(sess, user);
+		boolean found = false;
+		if (logins != null) {
+			found = logins.stream().anyMatch(attr -> namespace.equals(attr.getFriendlyNameParameter()));
+		}
+		if (!found) {
+			throw new InternalErrorException(user.toString() + " doesn't have login in namespace: " + namespace);
+		}
+
+		String subject;
+		try {
+			String attrName = ":usernameReminderMailSubject:" + namespace;
+			subject = getEmailTemplateFromEntitylessAttribute(sess, attrName, language);
+		} catch (AttributeNotExistsException ex) {
+			//If attribute not exists, log it and use null instead - default template for subject will be used
+			log.error("There is missing attribute with subject template for password reset in specific namespace.", ex);
+			subject = null;
+		}
+
+		String message;
+		try {
+			String attrName = ":usernameReminderMailTemplate:" + namespace;
+			message = getEmailTemplateFromEntitylessAttribute(sess, attrName, language);
+		} catch (AttributeNotExistsException ex) {
+			//If attribute not exists, log it and use null instead - default template for message will be used
+			log.error("There is missing attribute with message template for password reset in specific namespace.", ex);
+			message = null;
+		}
```

Comment:
Since the try block is identical for every email method, how about extracting it to the new private method as well? (we pass null along anyways so no need for handling in this method)

Reply:
Done as suggested.